### DISCUSSION TITLE
signout user when closed tab in browser or closed browser

### DISCUSSION THREAD

**Discussion by piotr-kozlowski-reps** [nextauthjs/next-auth#11936] [Help]
Hi,

I've got a problem and I couldn't find any solution to, so ... Here it is:

[ "next": "14.2.4"  "next-auth": "4.24.7" ]

I want to signOut when client closes tab or browser, I added a small provider, and wrapped app with it:

```
'use client';

import { signOut } from 'next-auth/react';
import { ReactNode, useEffect } from 'react';

type TProps = {
  children: ReactNode;
};

export function BrowserEventsProvider({ children }: TProps) {
  useEffect(() => {
    window.addEventListener('beforeunload', () => {
      signOut();
    });
  }, []);

  ////tsx
  return <>{children}</>;
}
```

But when I go back, session is still active and I got into app without being asked for login.

Looking forward to your help to achieve desired functionality, please
Regards
Piotr

**Comment by G0WTHAMRAJ** [comment]
did you find the solution i am facing the same issue

**Reply by piotr-kozlowski-reps** [reply]
Nope, sorry. I've given up. 
Client was told that's technically impossible. 
But I need to face it ... I screw up and had no time to dig further.

**Reply by G0WTHAMRAJ** [reply]
thanks for reply
